Update dependencies

valeriangalliat opened the pull request.

dustinblackman commented:
A lot of the rules changed between both versions of eslint, test a few of the integrations first before merging to make sure it doesn’t complain for anything silly. Then I'm good. 👊

valeriangalliat commented:
Yep I could find some additions that does not match our code, some I fixed the code 'cause it was edge cases and I thought it made sense, other I just removed the rule.

Once lint passes on all integrations I'll merge.

valeriangalliat commented:
Lint passed on all integrations!
